mypyc/ops.py

```diff
@@ -1733,6 +2010,45 @@ def __init__(
         self.classes = classes
         self.final_names = final_names
 
+    def serialize(self) -> JsonDict:
+        return {
+            'fullname': self.fullname,
+            'imports': self.imports,
+            'functions': [f.serialize() for f in self.functions],
+            'classes': [c.serialize() for c in self.classes],
+            'final_names': [(k, t.serialize()) for k, t in self.final_names],
+        }
+
+    @classmethod
+    def deserialize(cls, data: JsonDict, ctx: DeserMaps) -> 'ModuleIR':
+        return ModuleIR(
+            data['fullname'],
+            data['imports'],
+            [ctx.functions[FuncDecl.get_name_from_json(f['decl'])] for f in data['functions']],
+            [ClassIR.deserialize(c, ctx) for c in data['classes']],
+            [(k, deserialize_type(t, ctx)) for k, t in data['final_names']],
+        )
+
+
+def deserialize_scc(data: Dict[str, JsonDict], ctx: DeserMaps) -> Dict[str, ModuleIR]:
```

Comment:
Add docstring. Explain that this populates `ctx`, and that this expects that `ctx` contains information from all dependencies as a pre-condition?